### ISSUE TITLE
Publish job fail

### ISSUE THREAD

**Issue by Nikolara97** [aaronparker/intune-backup-azure#1]
Hi Aaron,

awesome article, I am just facing problem that I can't solve in last job (Publish) with fallowing error:

<img width="689" alt="image" src="https://user-images.githubusercontent.com/121571467/211324138-8e4646c7-c19a-4896-b00c-867f6fb0cc78.png">

Can you please help me with this? Whole raw error is below:

`2023-01-09T13:51:33.3533378Z ##[section]Starting: Convert markdown to HTML
2023-01-09T13:51:33.3540112Z ==============================================================================
2023-01-09T13:51:33.3540345Z Task         : Bash
2023-01-09T13:51:33.3540552Z Description  : Run a Bash script on macOS, Linux, or Windows
2023-01-09T13:51:33.3540741Z Version      : 3.214.0
2023-01-09T13:51:33.3540912Z Author       : Microsoft Corporation
2023-01-09T13:51:33.3541169Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/bash
2023-01-09T13:51:33.3541457Z ==============================================================================
2023-01-09T13:51:33.7210936Z Generating script.
2023-01-09T13:51:33.7227501Z Script contents:
2023-01-09T13:51:33.7229202Z cat "/home/vsts/work/1/s/prod-as-built.md" | md-to-pdf --config-file "/home/vsts/work/1/s/md2pdf/htmlconfig.json" --as-html > "/home/vsts/work/1/s/prod-as-built.html"
2023-01-09T13:51:33.7243060Z ========================== Starting Command Output ===========================
2023-01-09T13:51:33.7259787Z [command]/usr/bin/bash /home/vsts/work/_temp/8ab95b96-6f8c-4a7c-95ca-b330e27e1139.sh
2023-01-09T13:51:34.1968327Z Warning: couldn't read config file: /home/vsts/work/1/s/md2pdf/htmlconfig.json
2023-01-09T13:51:34.1988614Z Error: Cannot find module '/home/vsts/work/1/s/md2pdf/htmlconfig.json'
2023-01-09T13:51:34.1989436Z Require stack:
2023-01-09T13:51:34.1990438Z - /usr/local/lib/node_modules/md-to-pdf/dist/cli.js
2023-01-09T13:51:34.1995455Z     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:985:15)
2023-01-09T13:51:34.1997935Z     at Function.Module._load (node:internal/modules/cjs/loader:833:27)
2023-01-09T13:51:34.1999947Z     at Module.require (node:internal/modules/cjs/loader:1057:19)
2023-01-09T13:51:34.2002852Z     at require (node:internal/modules/cjs/helpers:103:18)
2023-01-09T13:51:34.2003362Z     at main (/usr/local/lib/node_modules/md-to-pdf/dist/cli.js:87:32)
2023-01-09T13:51:34.2003751Z     at processTicksAndRejections (node:internal/process/task_queues:96:5) {
2023-01-09T13:51:34.2004141Z   code: 'MODULE_NOT_FOUND',
2023-01-09T13:51:34.2004512Z   requireStack: [ '/usr/local/lib/node_modules/md-to-pdf/dist/cli.js' ]
2023-01-09T13:51:34.2004767Z }
2023-01-09T13:51:40.7252262Z ##[error]Bash wrote one or more lines to the standard error stream.
2023-01-09T13:51:40.7268637Z ##[error]Warning: couldn't read config file: /home/vsts/work/1/s/md2pdf/htmlconfig.json

2023-01-09T13:51:40.7272208Z ##[error]Error: Cannot find module '/home/vsts/work/1/s/md2pdf/htmlconfig.json'
Require stack:
- /usr/local/lib/node_modules/md-to-pdf/dist/cli.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:985:15)
    at Function.Module._load (node:internal/modules/cjs/loader:833:27)
    at Module.require (node:internal/modules/cjs/loader:1057:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at main (/usr/local/lib/node_modules/md-to-pdf/dist/cli.js:87:32)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/usr/local/lib/node_modules/md-to-pdf/dist/cli.js' ]
}

2023-01-09T13:51:40.7278484Z ##[section]Finishing: Convert markdown to HTML`

**Comment by aaronparker** [comment]
Have you copied across the `/md2pdf` directory? The error there is reporting that `/md2pdf/htmlconfig.json` is missing.

**Comment by Nikolara97** [comment]
Yeah that fixed my problem tnx 😄 Now all four stages are executed without any error, but pdf file is missing? After successful execution the whole intune backup is in directory, but no pdf at all. Artifacts are empty as well (no pdf there as well, like its mentioned in doc)... Any suggestion? Do I need to configure something additionally because i saw there is no any connector on our AzureDevops artifact part...
    On Monday, January 9, 2023 at 10:07:12 PM GMT+1, Aaron Parker ***@***.***> wrote:  
 
 


Have you copied across the /md2pdf directory? The error there is reporting that /md2pdf/htmlconfig.json is missing.

**Comment by aaronparker** [comment]
The PDF should be published as an artefact on the completed pipeline. View the pipeline summary and you'll see `2 published` that you can click on to then download the HTML or PDF version of the report.

The pipeline doesn't commit the PDF to the repo because PDF files won't diff that well. You could modify the pipeline to commit the HTML version of the report back to the repo instead.

**Comment by Nikolara97** [comment]
Thanks yeah, those pdfs were there :) I just have one last question. I noticed once pipeline is runed automatically on the schedule you defined in yaml file it always fails with following error. However, if I run in manually (Click on run pipeline) there is no any error at all. How can I solve this? Error is in job -> publish -> pull origin part.



    On Monday, January 9, 2023 at 10:43:25 PM GMT+1, Aaron Parker ***@***.***> wrote:  
 
 


The PDF should be published as an artefact on the completed pipeline. View the pipeline summary and you'll see 2 published that you can click on to then download the HTML or PDF version of the report.

The pipeline doesn't commit the PDF to the repo because PDF files won't diff that well. You could modify the pipeline to commit the HTML version of the report back to the repo instead.

**Comment by aaronparker** [comment]
I don't see the error posted to the thread.

**Comment by Nikolara97** [comment]
Oh sorry, my bad. Here it is.

<img width="524" alt="image" src="https://user-images.githubusercontent.com/121571467/212186901-84de5cdb-1773-4fb4-ab8c-ba60a8f9a1a1.png">

**Comment by Nikolara97** [comment]
@aaronparker any update regarding this issue?

**Comment by aaronparker** [comment]
Looks like you have changes in the repo while the pipeline is running. Does this happen everytime or just the once?

**Comment by Nikolara97** [comment]
This happens everytime (everyday it runs automatically), even if there are no changes to repo at all. And if i run it manually immediately after auto run is done, no errors or problem's at all...
    On Saturday, January 28, 2023 at 07:33:09 AM GMT+1, Aaron Parker ***@***.***> wrote:  
 
 


Looks like you have changes in the repo while the pipeline is running. Does this happen everytime or just the once?

**Comment by aaronparker** [comment]
I'm not sure what's going on then. Check the previous steps to ensure they're completing and writing back to the repo OK. You could try removing the `Pull origin` step in the `Publish` job to see whether the publish of the artefacts completes anyway.

**Comment by Nikolara97** [comment]
After removing pull origin step it's working without problems for 3 days in a row now. Thanks so much :) 
    On Sunday, January 29, 2023 at 12:50:45 AM GMT+1, Aaron Parker ***@***.***> wrote:  
 
 


I'm not sure what's going on then. Check the previous steps to ensure they're completing and writing back to the repo OK. You could try removing the Pull origin step in the Publish job to see whether the publish of the artefacts completes anyway.

**Comment by florian-ried** [comment]
> Oh sorry, my bad. Here it is.
> 
> <img alt="image" width="524" src="https://user-images.githubusercontent.com/121571467/212186901-84de5cdb-1773-4fb4-ab8c-ba60a8f9a1a1.png">

 I get the exact same error when I run the pipeline for the very first time. When I comment out the pull origin step in the publish area, I get the following error:

<img width="969" alt="image" src="https://user-images.githubusercontent.com/95634472/221657216-86145ba0-60a2-46a4-bee2-6c323a4a65c6.png">


Thank you so much :)
